tests/test_processing_suffixes.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from typing import Optional
+
+import pytest
+
+from unblob.processing import ExtractionConfig, process_file
+from unblob.report import OutputDirectoryExistsReport
+from unblob.testing import check_output_is_the_same
+
+TEST_DATA_PATH = Path(__file__).parent / "files/suffixes"
+
+
+def _patch(extraction_config: ExtractionConfig, carve_suffix, extract_suffix):
+    extraction_config.keep_extracted_chunks = False
+    if carve_suffix is not None:
+        extraction_config.carve_suffix = carve_suffix
+    if extract_suffix is not None:
+        extraction_config.extract_suffix = extract_suffix
+
+
+@pytest.mark.parametrize(
+    "carve_suffix,extract_suffix,expected_output_dir_name",
+    [
+        (None, None, "defaults"),
+        ("_c", "_e", "_c_e"),
+        ("_carve", "_extract", "_carve_extract"),
+    ],
+)
+def test_top_level_carve(
+    carve_suffix: Optional[str],
+    extract_suffix: Optional[str],
+    expected_output_dir_name: str,
+    extraction_config: ExtractionConfig,
+):
+    _patch(extraction_config, carve_suffix, extract_suffix)
+    input_file = TEST_DATA_PATH / "chunks"
+    carve_dir_name = input_file.name + extraction_config.carve_suffix
+    extract_dir_name = input_file.name + extraction_config.extract_suffix
+    expected_output_dir = (
+        TEST_DATA_PATH / "extractions" / expected_output_dir_name / carve_dir_name
+    )
```

Comment:
It feels something's hidden here. How is unblob made to extract into `expected_output_dir_name` ? Shouldn't unblob extract into `chunks_c` or `chunks_carve` as top-level ? What am I missing ?

Reply:
Yep, I was not happy with this either. Changed the input-output layout and renamed the variable to avoid confusion.